### ISSUE TITLE
style-prop-object does not allow null

### ISSUE THREAD

**Issue by vdh** [yannickcr/eslint-plugin-react#812]
I use `style` variables that are conditionally set as objects only when required, and AFAIK it's valid to provide null props to React.

**Comment by ljharb** [comment]
indeed, `null` and `undefined` should both be accepted.

**Comment by lencioni** [comment]
This might be worth looking into at the same time as #809

**Comment by vdh** [comment]
@ljharb This fails in `6.9.0` on an edge case of an explicit null variable:
```jsx
let style = null;
<div style={style}></div>
```
I dropped the unnecessary `= null` in my own code, so it's not an issue for me, but I thought I'd make a note of it just in case.

**Comment by ljharb** [comment]
@vdh fails in what way?

**Comment by vdh** [comment]
@ljharb [Line 37](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/style-prop-object.js#L37) doesn't have the same null check as [line 72](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/style-prop-object.js#L72), so it gives a `Style prop value must be an object` error.

**Comment by vdh** [comment]
I was also able to get a test failure by duplicating [this existing test](https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/style-prop-object.js#L120) and adding `= null` to it.

**Comment by ljharb** [comment]
@vdh thanks, this is now fixed in latest master.
